src/memjs/utils.ts

```diff
@@ -144,10 +144,19 @@ export interface Message {
   extras: Buffer;
 }
 
+const ERROR_TOO_MANY_OPEN_CONNECTIONS = "ERROR Too many open connections\r\n";
+
 export const parseMessage = function (dataBuf: Buffer): Message | false {
   if (dataBuf.length < 24) {
     return false;
   }
+
+  if (dataBuf.length === ERROR_TOO_MANY_OPEN_CONNECTIONS.length) {
```

Comment:
is there any known way we would get a partial buffer?

Reply:
it seems unlikely. I don't think a 33 byte write that would be the only thing the server writes would get split.

Even assuming that happens ,the cases are:

1. parseMessage gets a chunk of "ERROR Too many", incomplete message so waits for more data,  and then another chunk of " open connections\r\n" = handles it normally
2. we get a partial message and wait for more that never comes = command timeout is hit
3. we get a partial message and the socket is closed = newly fixed closed handler is hit.